deploy.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash 
+
+set -o errexit
+set -o pipefail
+set -vx
+
+# build a deployment package from a git repository
+
+# We assume that
+# - this script is in the repository which has been checked out into a working directory
+# - there is a .tool-versions file present for asdf
+# - the following tools are installed and on the path
+#   - asdf in /opt/asdf
+#   - node/npm (possibly via asdf)
+# - user-mode systemd is possible, and the user has the "linger-mode" enabled
+# - it is running as the user which will run the service, with a dbus session
+# - the following environment variables
+#   - DEPLOY_DEST
+#   - WWW_ROOT # not used if not writing proxy/path config
+#   - USERDIR: the current user's home directory
+#   - PROXY_PORT: the port number being proxied (e.g. '4000')
+#   - PROXY_PATH: the path being proxied (e.g. '/api')
+#   - BASE_URL_PATH # not actually used but might be in future?
+#   - GLITCHTIP_KEY
+#   - MAPTILER_API_KEY
+#   - DBUS_SESSION_BUS_ADDRESS: required for service management
+# - FIXME the user can write to DEPLOY_DEST, WWW_ROOT
+#   - or else, the proxy pass and WWW roots have been defined already
+#
+# We don't assume a terminal or any interactivity.
+
+FE_DEST=$DEPLOY_DEST/front-end
+#BE_DEST=$DEPLOY_DEST/back-end
+BE_DEST=$PWD/apps/back-end # work around build glitches by running from source!
+DATA_DEST=$DEPLOY_DEST/data
+SYSTEMD_UNIT="$USERDIR/.config/systemd/user/mykomap-backend.service"
+[[ -z "$DBUS_SESSION_BUS_ADDRESS" ]] && {
+  die "no user session found? DBUS_SESSION_BUS_ADDRESS unset."
+}
+
+mkdir -vp "${DEPLOY_DEST:?}"
+
+
+mkdir -vp "${FE_DEST}" "${BE_DEST}" "${DATA_DEST}"
+
+
+# update asdf installations, using .tool-versions file
+asdf plugin add nodejs || true
+asdf install
+
+# Install systemd unit
+mkdir -p "${SYSTEMD_UNIT%/*}"
+cat >"$SYSTEMD_UNIT" <<EOF
+[Unit]
+Description=Mykomap back-end process manager for %u
+Documentation=https://github.com/DigitalCommons/mykomap-monolith/
+After=network.target
+
+[Service]
+Type=exec
+LimitNOFILE=infinity
```

Comment:
The unit file for the PBS had these `=infinity` lines, I think since I copied your template from the ansible docs, but I removed them since they were leading to the disk running out of space (e.g. due to systemd log files or service's memory usage running out of control). Maybe we want to remove them here too?

Reply:
Removed them in this commit
https://github.com/DigitalCommons/mykomap-monolith/commit/f39cdc34543cb1cbdbc3682d1337d35f7e580e3e